vulnerabilities/importers/mozilla.py

```diff
@@ -0,0 +1,164 @@
+
+    def to_advisories(self, path: str) -> List[Advisory]:
+        """
+        Convert a file to corresponding advisories.
+        This calls proper method to handle yml/md files.
+        """
+        mfsa_id = self.mfsa_id_from_filename(path)
+
+        with open(path) as lines:
+            if path.endswith(".md"):
+                return self.get_advisories_from_md(mfsa_id, lines)
+            if path.endswith(".yml"):
+                return self.get_advisories_from_yml(mfsa_id, lines)
+
+        return []
+
+    def get_advisories_from_yml(self, mfsa_id, lines) -> List[Advisory]:
+        advisories = []
+        data = yaml.safe_load(lines)
+        data["mfsa_id"] = mfsa_id
+
+        fixed_package_urls = self.get_package_urls(data.get("fixed_in"))
+        references = self.get_yml_references(data)
+
+        if not data.get("advisories"):
+            return []
+
+        for cve, advisory in data["advisories"].items():
+            advisories.append(
+                Advisory(
+                    summary=advisory.get("description"),
+                    vulnerability_id=cve if is_cve(cve) else "",
+                    impacted_package_urls=[],
+                    resolved_package_urls=fixed_package_urls,
+                    references=references,
+                )
+            )
+
+        return advisories
+
+    def get_advisories_from_md(self, mfsa_id, lines) -> List[Advisory]:
+        yamltext, mdtext = split_markdown_front_matter(lines.read())
+        data = yaml.safe_load(yamltext)
+        data["mfsa_id"] = mfsa_id
+
+        fixed_package_urls = self.get_package_urls(data.get("fixed_in"))
+        references = self.get_yml_references(data)
+        cves = re.findall(r"CVE-\d+-\d+", yamltext + mdtext, re.IGNORECASE)
+        for cve in cves:
+            references.append(
+                Reference(
+                    reference_id=cve, url=f"https://cve.mitre.org/cgi-bin/cvename.cgi?name={cve}"
+                )
+            )
+
+        description = self.html_get_p_under_h3(markdown(mdtext), "description")
+
+        return [
+            Advisory(
+                summary=description,
+                vulnerability_id="",
+                impacted_package_urls=[],
+                resolved_package_urls=fixed_package_urls,
+                references=references,
+            )
+        ]
+
+    def html_get_p_under_h3(self, html, h3: str):
+        soup = BeautifulSoup(html, features="lxml")
+        h3tag = soup.find("h3", text=lambda txt: txt.lower() == h3)
+        p = ""
+        if h3tag:
+            for tag in h3tag.next_siblings:
+                if tag.name:
+                    if tag.name != "p":
+                        break
+                    p += tag.get_text()
+        return p
+
+    def mfsa_id_from_filename(self, filename):
+        match = MFSA_FILENAME_RE.search(filename)
+        if match:
+            return "mfsa" + match.group(1)
+
+        return None
```

Comment:
Does this even happen ? If there's no code to handle that

Reply:
As per the [given format[(https://github.com/mozilla/foundation-security-advisories#markdown-format) it should never happen.
Just in case it happens, I've now added to skip that file.